### PR TITLE
Fix BadWindow crash on Legacy graphics systems

### LINKED ISSUE #18061: Game crash while starting (BadWindow (invalid Window parameter))
Hi guys!

## System Information

- Operating System:  Lubuntu 18.04 (fresh install)
- OpenRA Version: `release-20200503`
- OpenRA Mod: Red Alert and Tiberian Dawn
- Source: Official AppImage

## Exception log

```
epoc@Laptop-Epoc:~/Documents$ chmod a+x OpenRA-Red-Alert-x86_64.AppImage
epoc@Laptop-Epoc:~/Documents$ ./OpenRA-Red-Alert-x86_64.AppImage
Platform is Linux
Engine version is release-20200503
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  4 (X_DestroyWindow)
  Resource id in failed request:  0x360001d
  Serial number of failed request:  281
  Current serial number in output stream:  282
Red Alert has encountered a fatal error.
Please refer to the crash logs and FAQ for more information.

Log files are located in ~/.openra/Logs
The FAQ is available at http://wiki.openra.net/FAQ
```

Same error log for Tiberian Dawn.

## Additional information

The aforementioned logs location (`~/.openra/Logs`) doesn't exist. I found them in `~/.config/openra/Logs/`, however everything's empty:

```
epoc@Laptop-Epoc:~/Documents$ ll ~/.config/openra/Logs/
total 8
drwxrwxr-x 2 epoc epoc 4096 mai    8 21:43 ./
drwxrwxr-x 3 epoc epoc 4096 mai    8 21:43 ../
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 debug.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 geoip.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 graphics.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 nat.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 perf.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 server.log
-rw-rw-r-- 1 epoc epoc    0 mai    9 11:09 sound.log
```

### PR DESCRIPTION
**Pull request by bggardner**
Resolves: #18061

### CODE REVIEW

**Comment by pchote** [comment]
Can we please have a comment somewhere, maybe before the new `SDL_Init` location, that `CanCreateGLWindow` now creates and destroys its own SDL contexts as a workaround for specific buggy drivers?

**Comment by pchote** [comment]
Please take a look at https://github.com/OpenRA/OpenRA/wiki/Contributing#updating-a-pull-request for instructions on how to fix the excess merge commits.

File: OpenRA.Platforms.Default/Sdl2PlatformWindow.cs [hunk 1]
```diff
@@ -161,6 +159,10 @@ public GLProfile[] SupportedGLProfiles
 					throw new InvalidOperationException("No supported OpenGL profiles were found.");
 
 				profile = supportedProfiles.Contains(requestProfile) ? requestProfile : supportedProfiles.First();
+
+				// Note: This must be called after the CanCreateGLWindow checks above,
+                // which needs to create and destroy its own SDL contexts as a workaround for specific buggy drivers
```
Comment (abcdefg30) [inline OpenRA.Platforms.Default/Sdl2PlatformWindow.cs]: Please use tabs instead of spaces.

**Comment by pchote** [comment]
Here are test builds to verify that the fix works: https://github.com/pchote/OpenRA/releases/tag/devtest-20200829

**Comment by bggardner** [comment]
Looks good to me, even with the system `libSDL2` libraries uninstalled.

**Review by pchote — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6c364ac47c1d7d787e40566af4aa21cbe5b606fe...ee3981f90210a7cf23f4174797c0a0e7bcf3df6e)

**Pull request merged**